src/wallet/wallet2.cpp

```diff
@@ -4236,6 +4410,373 @@ static uint32_t get_count_above(const std::vector<wallet2::transfer_details> &tr
+  cryptonote::COMMAND_RPC_LIGHT_WALLET_GET_UNSPENT_OUTS::response ores;
+  
+  oreq.amount = "0";
+  oreq.address = get_account().get_public_address_str(m_testnet);
+  oreq.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/get_unspent_outs", oreq, ores, m_http_client, std::chrono::seconds(5), "POST");
+  m_daemon_rpc_mutex.unlock();
+  if (!r) {
+    MERROR("Lightwallet: Failed to get unspent outs");
+    return;
+  }
+  
+
+  
+  std::unordered_map<crypto::hash,bool> transfers_txs;
+  for(const auto &t: m_transfers)
+    transfers_txs.emplace(t.m_txid,t.m_spent);
+  
+  MDEBUG("FOUND " << ores.outputs.size() <<" outputs");
+  
+  for (const auto &o: ores.outputs) {
+    bool spent = false;
+    bool add_transfer = true;
+    crypto::key_image unspent_key_image;
+    crypto::public_key tx_public_key = AUTO_VAL_INIT(tx_public_key);
+    string_tools::hex_to_pod(o.tx_pub_key, tx_public_key);
+    
+    for (const std::string &ski: o.spend_key_images) {
+      spent = false;
+
+      // Check if key image is for real
+      crypto::key_image calculated_key_image;
+      cryptonote::keypair in_ephemeral;
+      //TODO: Add validation: key_imageIsNotNull
+      string_tools::hex_to_pod(ski, unspent_key_image);
+
+      cryptonote::generate_key_image_helper(get_account().get_keys(), tx_public_key, o.index, in_ephemeral, calculated_key_image);
+      
+      if(unspent_key_image == calculated_key_image){
+        MDEBUG("Output " << o.public_key << " is spent. Key image: " <<  ski);
+        spent = true;
+        break;
+      } {
+        // MDEBUG("Unspent output " << o.public_key << " used as mixin with key image " << ski);
+      }
+    }
+
+    // Check if tx already exists in m_transfers. 
+    crypto::hash txid;
+    crypto::public_key tx_pub_key;
+    crypto::public_key public_key;
+    string_tools::hex_to_pod(o.tx_hash, txid);
+    string_tools::hex_to_pod(o.public_key, public_key);
+    string_tools::hex_to_pod(o.tx_pub_key, tx_pub_key);
+    
+
+    for(auto &t: m_transfers){
+      if(t.get_public_key() == public_key) {
+        t.m_spent = spent;
+        MTRACE(string_tools::pod_to_hex(public_key) << " already in m_transfers - Not adding");
+        add_transfer = false;
+        break;
+      }
+    }
+    
+    if(!add_transfer)
+      continue;
+    
+    MTRACE("Adding output " << o.public_key);
+    m_transfers.push_back(boost::value_initialized<transfer_details>());
+    transfer_details& td = m_transfers.back();
+    
+    td.m_block_height = o.height;
+    td.m_global_output_index = o.global_index;
+    td.m_txid = txid;
+    
+
+    
+    // Add to extra
+    add_tx_pub_key_to_extra(td.m_tx, tx_pub_key);
+    
+    td.m_key_image = unspent_key_image;
+    td.m_key_image_known = !m_watch_only;
+    td.m_amount = boost::lexical_cast<uint64_t>(o.amount);
+    td.m_pk_index = 0; //TODO: Not sure what this is?
```

Comment:
This is the index of the correct pubkey in tx extra, because there was a bug in cold signing where the cold wallet would add its pubkey to tx extra, but the hot wallet had already added one, so you'd end up with two, the first one being useless.